src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/typeresolution.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include <RoMetadataApi.h>
+#include <hstring.h>
+#include <atlbase.h>
+#include <windows.foundation.h>
+#include <Windows.h>
+#include "wil/result.h"
+#include "wil/resource.h"
+#include "wil/filesystem.h"
+#include "Synchapi.h"
+#include <unordered_map>
+
+#define METADATA_FILE_EXTENSION L"winmd"
+#define METADATA_FILE_PATH_FORMAT L"%s%s."  METADATA_FILE_EXTENSION
+#define METADATA_FILE_SEARCH_FORMAT L"%s%s*."  METADATA_FILE_EXTENSION
+
+namespace UndockedRegFreeWinRT
+{
+    // This number comes from (c_cchTypeNameMax = 512) in //depot/fbl_ie_dev1/com/WinRT/buildtools/mdmerge/MetadataScope.h
+    static const UINT32 g_uiMaxTypeName = 512;
```

Comment:
I don't see why this needs to be in the header. It's only used in the cpp implementation.

Reply:
Good point

Reply:
Moved a bunch of these to the cpp implementation per suggestion